`HasOne` doesn't compile

Code:
```rust
#[derive(Debug, toasty::Model)]
struct A {
    #[key]
    id: Uuid,

    #[has_one]
    b: toasty::HasOne<B>,
}

#[derive(Debug, toasty::Model)]
struct B {
    #[key]
    id: Uuid,

    #[belongs_to(key = a_id, references = id)]
    a: toasty::BelongsTo<A>,
    a_id: Uuid,
}
```

Error:
```
error[E0277]: HasOne requires the B::a field to be of type `BelongsTo<Self>`, but it was `toasty::BelongsTo<A>` instead
  --> src/features/user/mod.rs:31:5
   |
31 | /     #[has_one]
32 | |     b: toasty::HasOne<B>,
   | |________________________^ Has one associations require the target to include a back-reference
   |
   = help: the trait `user::_::<impl A>::b::Verify<B>` is not implemented for `toasty::BelongsTo<A>`
   = note: Note 1
note: required by a bound in `user::_::<impl A>::b::verify`
  --> src/features/user/mod.rs:31:5
   |
31 | /     #[has_one]
32 | |     b: toasty::HasOne<B>,
   | |________________________^ required by this bound in `verify`
```

--- a/crates/toasty-codegen/src/expand/relation.rs
+++ b/crates/toasty-codegen/src/expand/relation.rs
@@ -279,7 +279,10 @@ impl Expand<'_> {
             field_ident.span(),
         );
 
-        let my_msg = format!("HasMany requires the {{A}}::{pair_ident} field to be of type `BelongsTo<Self>`, but it was `{{Self}}` instead");
+        let verify_a = util::ident("A");
+        let verify_t = util::ident("T");
+
+        let my_msg = format!("HasMany requires the {{{verify_a}}}::{pair_ident} field to be of type `BelongsTo<Self>`, but it was `{{Self}}` instead");
         let my_label =
             "Has many associations require the target to include a back-reference".to_string();
 
@@ -287,8 +290,8 @@ impl Expand<'_> {
             // Reference the field to generate a compiler error if it is missing.
             #[allow(unreachable_code)]
             if false {
-                fn load<T: #toasty::Model>() -> T {
-                    T::load(todo!()).unwrap()
+                fn load<#verify_t: #toasty::Model>() -> #verify_t {
+                    #verify_t::load(todo!()).unwrap()
                 }
 
                 #[diagnostic::on_unimplemented(
@@ -297,18 +300,18 @@ impl Expand<'_> {
                     note = "Note 1",
                     // note = "Note 2"
                 )]
-                trait Verify<A> {
+                trait Verify<#verify_a> {
                 }
 
                 #[diagnostic::do_not_recommend]
-                impl<A> Verify<A> for #toasty::BelongsTo<#model_ident> {
+                impl<#verify_a> Verify<#verify_a> for #toasty::BelongsTo<#model_ident> {
                 }
 
                 #[diagnostic::do_not_recommend]
-                impl<A> Verify<A> for #toasty::BelongsTo<Option<#model_ident>> {
+                impl<#verify_a> Verify<#verify_a> for #toasty::BelongsTo<Option<#model_ident>> {
                 }
 
-                fn verify<T: Verify<A>, A>(_: &T) {
+                fn verify<#verify_t: Verify<#verify_a>, #verify_a>(_: &#verify_t) {
                 }
 
                 let instance = load::<<#ty as #toasty::Relation>::Model>();
@@ -343,7 +346,10 @@ impl Expand<'_> {
         let model_ident = &self.model.ident;
         let pair_ident = syn::Ident::new(&self.model.name.ident.to_string(), rel.span);
 
-        let my_msg = format!("HasOne requires the {{A}}::{pair_ident} field to be of type `BelongsTo<Self>`, but it was `{{Self}}` instead");
+        let verify_a = util::ident("A");
+        let verify_t = util::ident("T");
+
+        let my_msg = format!("HasOne requires the {{{verify_a}}}::{pair_ident} field to be of type `BelongsTo<Self>`, but it was `{{Self}}` instead");
         let my_label =
             "Has one associations require the target to include a back-reference".to_string();
 
@@ -351,8 +357,8 @@ impl Expand<'_> {
             // Reference the field to generate a compiler error if it is missing.
             #[allow(unreachable_code)]
             if false {
-                fn load<T: #toasty::Model>() -> T {
-                    T::load(todo!()).unwrap()
+                fn load<#verify_t: #toasty::Model>() -> #verify_t {
+                    #verify_t::load(todo!()).unwrap()
                 }
 
                 #[diagnostic::on_unimplemented(
@@ -361,18 +367,18 @@ impl Expand<'_> {
                     note = "Note 1",
                     // note = "Note 2"
                 )]
-                trait Verify<A> {
+                trait Verify<#verify_a> {
                 }
 
                 #[diagnostic::do_not_recommend]
-                impl<A> Verify<A> for #toasty::BelongsTo<#model_ident> {
+                impl<#verify_a> Verify<#verify_a> for #toasty::BelongsTo<#model_ident> {
                 }
 
                 #[diagnostic::do_not_recommend]
-                impl<A> Verify<A> for #toasty::BelongsTo<Option<#model_ident>> {
+                impl<#verify_a> Verify<#verify_a> for #toasty::BelongsTo<Option<#model_ident>> {
                 }
 
-                fn verify<T: Verify<A>, A>(_: T) {
+                fn verify<#verify_t: Verify<#verify_a>, #verify_a>(_: #verify_t) {
                 }
 
                 let instance = load::<<#ty as #toasty::Relation>::Model>();

